storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -154,13 +154,15 @@ func (s *commitLogSource) Read(
 		// datapoints for a given shard/series will be processed in a serialized
 		// manner.
 		encoderChans[series.Shard%uint32(numConc)] <- encoderArg{
```

Comment:
i wonder if `series.UniqueIndex % ...` would help distribute the load better?

Reply:
yeah agreed

Reply:
Ok actually, no I remember why I did this now. Distributing by shard means I can avoid a lock because they're never contending over the same index in the unmerged slice. Removing a lock contention is probably worth potentially uneven load. I'll add a comment so this is more clear.